src/corecel/math/ArrayUtils.hh

```diff
@@ -124,6 +130,24 @@ CELER_FUNCTION T norm(Array<T, N> const& v)
     return std::sqrt(dot_product(v, v));
 }
 
+//---------------------------------------------------------------------------//
+/*!
+ * Construct a unit vector.
+ *
+ * Unit vectors have an Euclidian norm magnitude of 1.
+ */
+template<class T, size_type N>
```

Comment:
Would this make more sense as an in-place operation, since there is already an existing array?

Reply:
The `normalize_direction` already does that, but the reason for making this function is because it's much nicer to be able to construct something with an rvalue than to have to execute a second statement to modify an lvalue. For example:
```c++
EXPECT_FALSE(softeq_(ref, Plane{make_unit_vector(Real3{-1, 0, 1}), 2.0}));
```
rather than the existing
```c++
Real3 normal{-1, 0, 1};
normalize_direction(&normal);
EXPECT_FALSE(softeq_(ref, Plane{normal, 2.0}));
```
which also pollutes the rest of the scope with an unused mutable variable.

It looks like `y = make_unit_vector(y)` is identically equivalent to `normalize_direction(&y)` so I actually think I'll replace the implementation of the latter with the former. In a separate PR I can delete `normalize_direction`.